api/envoy/api/v2/core/health_check.proto

```diff
@@ -95,12 +96,20 @@ message HealthCheck {
     repeated string request_headers_to_remove = 8;
 
     // If set, health checks will be made using http/2.
-    bool use_http2 = 7;
+    // Deprecated, use :ref:`codec_client_type
+    // <envoy_api_field_core.HealthCheck.HttpHealthCheck.codec_client_type>` instead.
+    bool use_http2 = 7 [deprecated = true];
 
     // Specifies a list of HTTP response statuses considered healthy. If provided, replaces default
     // 200-only policy - 200 must be included explicitly as needed. Ranges follow half-open
     // semantics of :ref:`Int64Range <envoy_api_msg_type.Int64Range>`.
     repeated type.Int64Range expected_statuses = 9;
+
+    // Use specified application protocol for health checks. This is to replace
```

Comment:
I'd be inclined to remove everything after the first sentence, since we might forget to clean it up when we remove use_http2

Reply:
Actually the mutual reference is suggested by htuch. If we remove use_http2, I think the :ref: will be broken if we don't clean up.

Reply:
Oops sorry, I'd missed that.  Yeah, if the docs build would break if we don't clean up this is 100% a benefit as it's helpful while we have both and will go away when we don't